### PR TITLE
Ternary Sniff: remove whitelist approach to sniffs

### PR DESCRIPTION
**Pull request by mikesherov**

### CODE REVIEW

**Comment by bmax** [comment]
👍

File: tests/Behance/Sniffs/ControlStructures/TernarySniffTest.inc [hunk 1]
```diff
@@ -213,3 +213,7 @@ $adapter_options = ( !empty( $options[ $url ] ) )
 $options = ( $validator->options && is_array( $validator->options ) )
            ? $validator->options
            : [];
+
+$a = isset( $b )
```
Comment (andyroo2000) [inline tests/Behance/Sniffs/ControlStructures/TernarySniffTest.inc]: Oh... I thought we were trying to enforce wrapping parens on the first line of the statement?

This works for me... just making sure I understand the rule.

File: Behance/Sniffs/ControlStructures/TernarySniff.php [hunk 2]
```diff
@@ -49,19 +49,25 @@ public function process( PHP_CodeSniffer_File $phpcsFile, $stackPtr ) {
...
+    $this->_start_of_statement = $this->_phpcsFile->findStartOfStatement( $stackPtr );
+    $this->_close_of_parens    = $this->_phpcsFile->findPrevious( PHP_CodeSniffer_Tokens::$emptyTokens, ($stackPtr - 1), $this->_start_of_statement, true );
 
-    if ( $this->_isMultilineTernaryWithoutWrappingParenthesis() ) {
+    if ( $this->_tokens[ $this->_close_of_parens ]['code'] !== T_CLOSE_PARENTHESIS ) {
       $error = 'The first statement must be wrapped in parenthesis';
       return $this->_phpcsFile->addError( $error, $this->_stackPtr, $code );
     }
 
+    $this->_open_parenthesis = $this->_tokens[ $this->_close_of_parens ]['parenthesis_opener'];
```
Comment (andyroo2000) [inline Behance/Sniffs/ControlStructures/TernarySniff.php]: 🎉

**Comment by andyroo2000** [comment]
👍

File: tests/Behance/Sniffs/ControlStructures/TernarySniffTest.inc (hunk 1 above)
Comment (bmax) [inline tests/Behance/Sniffs/ControlStructures/TernarySniffTest.inc]: yes, you must wrap it with ( ); line `217` will throw an error!

**Pull request merged**